paths.go

```diff
@@ -232,3 +232,104 @@ func findSCC[K comparable](vertexHash K, state *sccState[K]) {
 		state.components = append(state.components, component)
 	}
 }
+
+// AllPathsBetweenTwoVertices list all paths from start to end.
+func AllPathsBetweenTwoVertices[K comparable, T any](g Graph[K, T], start, end K) ([][]K, error) {
+	adjacencyMap, err := g.AdjacencyMap()
+	if err != nil {
+		return nil, err
+	}
+
+	mainStack, viceStack := newStack[K](), newStack[stack[K]]()
+
+	checkEmpty := func() error {
+		if mainStack.isEmpty() || viceStack.isEmpty() {
+			return errors.New("empty stack")
+		}
+		return nil
+	}
+
+	buildLayer := func(element K) {
+		mainStack.push(element)
+
+		newElements := newStack[K]()
+		for e := range adjacencyMap[element] {
+			var contains bool
+			mainStack.forEach(func(k K) {
+				if e == k {
+					contains = true
+				}
+			})
+			if contains {
+				continue
+			}
+			newElements.push(e)
+		}
+		viceStack.push(newElements)
+	}
```

Comment:
I got the idea from this [blog](https://boycgit.github.io/all-paths-between-two-vertex/) (but I have not found the English-described version and the pictures in the blog has some mistakes).

I have renamed the function to `AllPathsBetween`.

I considered adding the method `stack.contains`. However, I'm not sure whether it's good or not to define the generic type of `stack` as `comparable` rather than `any`. Thus, I trickly used the method `forEach`. If a `contains` method is added to `stack`, this block of code will be nicer ~

Reply:
@tzq0301 Thanks for the link to the blog post!

> I considered adding the method `stack.contains`. However, I'm not sure whether it's good or not to define the generic type of `stack` as `comparable` rather than `any`.

I see. Since `stack` is only being used internally and only deals with hash values of type `K comparable` anyway, I think it is fine to constrain it to `comparable`.